storefront.rb

```diff
@@ -3,11 +3,123 @@
 require 'tty'
 require 'active_record'
 require 'sqlite3'
-require_relative 'password'
+require 'pry'
+require_relative 'models/user/'
+require_relative 'models/address/'
+require_relative 'models/item/'
+require_relative 'models/order/'
+require_relative 'models/review/'
+
+# ActiveRecord::Base.logger = Logger.new(STDOUT)
 
 ActiveRecord::Base.establish_connection(
   :adapter => 'sqlite3',
   :database => 'db/store.sqlite3'
 )
 
-# Your code here
+###### Explorer Mode
+puts"\n"
+# How many users are there?
+puts "There are #{User.count} users."
+
+# What are the 5 most expensive items?
+puts "The five most expensive items are:"
+Item.order(price: :desc).limit(5).each do |item|
+  puts " #{item.title} costs $#{item.price}"
+end
+puts "\n"
+
+# What's the cheapest book?
+cheap_book = Item.where("category LIKE '%books%'").order(:price).first
+puts "#{cheap_book.title} is the cheapest book at $#{cheap_book.price}\n\n"
+
+# Who lives at "6439 Zetta Hills, Willmouth, WY"? Do they have another address?
+zeta = Address.find_by(street: "6439 Zetta Hills", city: "Willmouth", state: "WY").user
+puts "#{zeta.first_name} #{zeta.last_name} lives at 6439 Zetta Hills, Willmouth, WY."
+
+puts "Corrine Little has two addresses:"
+zeta.addresses.each do |x|
+  puts "#{x.street}, #{x.city}, #{x.state}"
+end
+puts "\n"
+
+# Correct Virginie Mitchell's address to "New York, NY, 10108".
+
+virg = User.find_by(first_name: "Virginie", last_name: "Mitchell").addresses.find_by(state: "NY")
+virg.update(city: "New York", zip: 10108)
+puts "Fixed address to #{virg.street}, #{virg.city}, #{virg.state}, #{virg.zip}"
+
+# How much would it cost to buy one of each tool?
+total = Item.where("category LIKE '%tool%'").sum(:price)
+puts "Buying one of each tool costs $#{total}."
+puts "\n"
+# How many total items did we sell?
+total_items = Order.sum(:quantity)
+puts "We sold #{total_items} items."
+puts "\n"
+# How much was spent on books?
+spent = Item.joins(:orders).where("category LIKE '%books%'").sum("price*quantity")
+puts "$#{spent} was spent on books."
+
+######### Adventure Mode
+
+# What item was ordered most often? Grossed the most money?
+ most_order = Order.joins(:item).order("sum_orders_quantity desc").group(:title).sum("orders.quantity").first
+ puts "#{most_order.first} was ordered the most."
+
+ gross = Order.joins(:item).order("sum_orders_quantity_all_items_price desc").group(:title).sum("orders.quantity * items.price").first
+ puts "#{gross.first} grossed the most money."
+
+# What user spent the most?
+
+baller = Order.joins(:user, :item).order("sum_orders_quantity_all_items_price desc").group("users.id").sum("orders.quantity * items.price").first
+baller_name = User.where("id = ?", baller.first).first
+puts "#{baller_name.first_name} #{baller_name.last_name} spent the most money.\n\n"
+## There has to be a better way to do this
```

Comment:
:man_shrugging: ... could be refactored for a `full_name` method on users. Could also group by `user_id` on orders and not join :users